### PR TITLE
Artifact grouping take 2

### PR DESCRIPTION
**Pull request by mitchdenny**
This PR changes the layout of the files in the artifact directory from the default structure of a Maven repo to the following structure:
```<root>/<groupId>/<artifactId>```

For example:
```<root>/com.azure/azure-sdk-template```

This is to remove the need for downstream consumers of the artifacts to use wildcards to discover artifact locations which is potentially error prone. As a consequence I've removed the ```stage-artifacts.yml``` template from the repository, and anywhere it was used instead use a combination of ```${{artifact.groupId}}/${{artifact.name}}``` to locate the artifact instances (within an ```${{ each ... }}:``` section).

Additionally I've moved common logic that was previously in ```Publish-MavenPackages.ps1``` into a PowerShell module in ```eng/scripts/MavenPackaging```. This logic is then used by ```Publish-MavenPackages.ps1``` and a new script ```Stage-MavenPackageArtifacts.ps1```.

### CODE REVIEW

File: eng/pipelines/templates/stages/archetype-java-release-pom-only.yml [hunk 1]
```diff
@@ -66,14 +66,9 @@ stages:
...
                     - checkout: self
-                    - template: /eng/pipelines/templates/steps/stage-artifacts.yml
-                      parameters:
-                        SourceFolder: ${{parameters.ArtifactName}}-signed
-                        TargetFolder: ${{artifact.safeName}}
-                        PackageName: ${{artifact.name}}
                     - template: /eng/common/pipelines/templates/steps/create-tags-and-git-release.yml
                       parameters:
-                        ArtifactLocation: $(Pipeline.Workspace)/${{artifact.safeName}}
+                        ArtifactLocation: ${{parameters.ArtifactName}}-signed/${{artifact.groupId}}/${{artifact.name}}
```
Comment (JimSuplizio) [inline eng/pipelines/templates/stages/archetype-java-release-pom-only.yml:71]: @mitchdenny shouldn't the artifact location be `$(Pipeline.Workspace)/${{parameters.ArtifactName}}-signed/${{artifact.groupId}}/${{artifact.name}}` like it is in eng/pipelines/templates/stages/archetype-java-release.yml?

File: eng/pipelines/templates/stages/archetype-java-release.yml [hunk 2]
```diff
@@ -165,14 +160,9 @@ stages:
...
                       - checkout: self
-                      - template: /eng/pipelines/templates/steps/stage-artifacts.yml
-                        parameters:
-                          SourceFolder: ${{parameters.ArtifactName}}-signed
-                          TargetFolder: ${{artifact.safeName}}
-                          PackageName: ${{artifact.name}}
                       - template: /eng/common/pipelines/templates/steps/docs-metadata-release.yml
                         parameters:
-                          ArtifactLocation: $(Pipeline.Workspace)/${{artifact.safeName}}
+                          ArtifactLocation: ${{parameters.ArtifactName}}-signed/${{artifact.groupId}}/${{artifact.name}}
```
Comment (JimSuplizio) [inline eng/pipelines/templates/stages/archetype-java-release.yml:165]: Shouldn't this also be `$(Pipeline.Workspace)/${{parameters.ArtifactName}}-signed/${{artifact.groupId}}/${{artifact.name}}`?

File: eng/scripts/MavenPackaging/MavenPackaging.psd1 [hunk 3]
```diff
@@ -0,0 +1,18 @@
+@{
```
Comment (weshaggard) [inline eng/scripts/MavenPackaging/MavenPackaging.psd1:1]: Unless there is a good reason for this to be a PS module I'd suggest just having it be a helper script that we dot source into places that need it. PS modules end up causing more trouble then they are worth in a lot of our usages as such @chidozieononiwu is actually starting to move us away from them.

Comment (mitchdenny) [reply eng/scripts/MavenPackaging/MavenPackaging.psd1:1]: What kind of problems? Curious now.

File: eng/pipelines/templates/stages/archetype-java-release.yml (hunk 2 above)
Comment (mitchdenny) [reply eng/pipelines/templates/stages/archetype-java-release.yml:165]: Fixed.

Comment (mitchdenny) [reply eng/pipelines/templates/stages/archetype-java-release.yml:165]: Probably would have worked anyway but consistency is good.

File: eng/pipelines/templates/stages/archetype-java-release-pom-only.yml (hunk 1 above)
Comment (mitchdenny) [reply eng/pipelines/templates/stages/archetype-java-release-pom-only.yml:71]: Fixed

File: eng/scripts/MavenPackaging/MavenPackaging.psd1 (hunk 3 above)
Comment (mitchdenny) [reply eng/scripts/MavenPackaging/MavenPackaging.psd1]: I've moved back to dot-sourcing for now.

Comment (JimSuplizio) [reply eng/scripts/MavenPackaging/MavenPackaging.psd1]: @mitchdenny, if you recall a couple of weeks back, there were a bunch of [transient errors installing PS modules that Dozie had to fix](https://github.com/Azure/azure-sdk-tools/pull/781).

**Comment by mitchdenny** [comment]
Yep. Agreed.

File: eng/scripts/MavenPackaging/MavenPackaging.psd1 (hunk 3 above)
Comment (weshaggard) [reply eng/scripts/MavenPackaging/MavenPackaging.psd1]: Yes there are various transient networking errors that we can completely avoided if we use all local scripts. Even using local modules things are a little difficult to debug and test locally as you always have to remove the module after you make an updated change to it before you can reinstall it. There have also been other odd issues like we had to disable the progress dialog in various places when installing even local modules. There were enough small issues that I felt we weren't getting anything useful from PS modules so I've decided to just use raw scripts.

**Pull request merged**